mtls-southbound/README.md

```diff
@@ -0,0 +1,146 @@
+
+# create PCKS12 export, use 'test' as password
+openssl pkcs12 -export -out client.p12 -inkey key.pem -in cert.pem
+
+# verify certificate, should return OK
+openssl verify -CAfile cert.pem cert.pem
+```
+
+## Step 4: Set VM cert and nginx config
+
+Now we will sync the cert and key to the VM, and set the nginx config file as well.
+
+```sh
+# copy cert, key and nginx.conf files to VM /etc/nginx dir
+gcloud compute scp cert.pem key.pem nginx.conf $VM_NAME:/etc/nginx --zone=$ZONE --project $PROJECT_ID
+
+# restart nginx to apply config
+gcloud compute ssh $VM_NAME --zone=$ZONE --project=$PROJECT_ID --command="sudo nginx -s reload"
+```
+
+## Step 5: Test calling VM directly with cert and key
+
+Now that our nginx is running with our cert and key, we can call it using `curl`.
+
+```sh
+# call with just the cacert, should get the message "blocked access to mTLS-protected resource"
+curl -v https://$VM_IP --cacert cert.pem
+
+# now call with cert and key, should get the message "access to mTLS-protected resource"
+curl -v https://$VM_IP --cacert cert.pem --key key.pem --cert cert.pem
+```
+
+## Step 6: Create Apigee Keystore for cert and key to use in a proxy
+
+```sh
+# create Apigee keystore
+apigeecli keystores create -n mtls-keystore1 -e $APIGEE_ENV -o $PROJECT_ID -t $(gcloud auth print-access-token)
+
+# create Apigee key and cert
+apigeecli keyaliases create -s test-key1 -f keycertfile -k mtls-keystore1 --key-filepath key.pem --cert-filepath cert.pem -e $APIGEE_ENV -o $PROJECT_ID -t $(gcloud auth print-access-token)
+
+# create Apigee cert
+apigeecli keyaliases create -s test-cert1 -f keycertfile -k mtls-keystore1 --cert-filepath cert.pem -e $APIGEE_ENV -o $PROJECT_ID -t $(gcloud auth print-access-token)
+
+# create Apigee target server
+apigeecli targetservers create -c true -s "$VM_IP" -i true --keyalias test-key1 --keystore mtls-keystore1 -n mtls-service -p 443 --tls true --tlsenforce false --truststore mtls-keystore1 -e $APIGEE_ENV -o $PROJECT_ID -t $(gcloud auth print-access-token)
+
+# deploy Apigee proxy
+apigeecli apis create bundle -f apiproxy --name SecureBackendProxy-v1 -o $PROJECT_ID -e $APIGEE_ENV --ovr -t $(gcloud auth print-access-token)
+```
+
+## Step 7: Test Apigee API proxy
+
+```sh
+# get Apigee hostname, this will get the first one from the first envgroup, if not correct then adjust..
+HOSTNAME=$(apigeecli envgroups list -o $PROJECT_ID | jq --raw-output '.environmentGroups[0].hostnames[0]')
+
+# call Apigee API proxy
+curl https://$HOSTNAME/v1/samples/mtls-service
+# you should get back the message "access to mTLS-protected resource" since Apigee has the mTLS cert and key. Yay!
+```
+
+## Step 8 (Optional): Test nginx service locally with Docker
```

Comment:
Why is this needed? What am I doing within this image? I dont see the steps to use this image in the VM

Reply:
It's just a nice to have command to test the nginx locally, no big deal, just a small extra for testing purposes.. updated docs to explain that.